src/java/picard/sam/util/ReadNameParsingUtils.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.sam.util;
+
+/**
+ * Common functions for quickly parsing strings. Used for parsing the tile and coordinates from the read names
+ */
+public class ReadNameParsingUtils {
+
+    /**
+     * Single pass method to parse the read name for the default regex.  This will only insert the 2nd to the 4th
+     * tokens (inclusive).  It will also stop after the fifth token has been successfully parsed.
+     */
+    static public int getRapidDefaultReadNameRegexSplit(final String readName, final char delim, final int[] tokens) {
+        int tokensIdx = 0;
+        int prevIdx = 0;
+        for (int i = 0; i < readName.length(); i++) {
+            if (readName.charAt(i) == delim) {
+                if (1 < tokensIdx && tokensIdx < 5)
+                    tokens[tokensIdx] = rapidParseInt(readName.substring(prevIdx, i)); // only fill in 2-4 inclusive
+                tokensIdx++;
+                if (4 < tokensIdx) return tokensIdx; // early return, only consider the first five tokens
+                prevIdx = i + 1;
+            }
+        }
+        if (prevIdx < readName.length()) {
+            if (1 < tokensIdx && tokensIdx < 5)
+                tokens[tokensIdx] = rapidParseInt(readName.substring(prevIdx, readName.length())); // only fill in 2-4 inclusive
+            tokensIdx++;
+        }
+        return tokensIdx;
+    }
+
+    /**
+     * Very specialized method to rapidly parse a sequence of digits from a String up until the first
+     * non-digit character.
+     */
+    static public int rapidParseInt(final String input) {
```

Comment:
Does the use of this method result in significantly more performant code than a `-?\d+` regex and `Integer.valueOf`?  Same question for the method above.